src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs

```diff
@@ -204,6 +207,15 @@ public void QueueTimelineRecordUpdate(Guid timelineId, TimelineRecord timelineRe
             _timelineUpdateQueue[timelineId].Enqueue(timelineRecord.Clone());
         }
 
+        public void ReportThrottling(string delay, string expiration)
+        {
+            Trace.Info($"Receive server throttling report, expect delay {delay} milliseconds till {expiration}");
+            if (JobServerQueueThrottling != null)
```

Comment:
Usually a better pattern to assign the event to a local variable, then use the local variable for null check and firing the event. This way if the property is set to null in between the null check and calling the method, you do not NullReferenceException. :-)